packages/iov-ethereum/src/ethereumconnection.ts

```diff
@@ -53,16 +82,25 @@ async function loadChainId(baseUrl: string): Promise<ChainId> {
 }
 
 export class EthereumConnection implements BcpConnection {
-  public static async establish(baseUrl: string): Promise<EthereumConnection> {
+  public static async establish(baseUrl: string, wsUrl: string | undefined): Promise<EthereumConnection> {
     const chainId = await loadChainId(baseUrl);
-    return new EthereumConnection(baseUrl, chainId);
+    const connection = new EthereumConnection(baseUrl, chainId, wsUrl);
+    if (connection.socket) {
+      await connection.socket.connected;
```

Comment:
Can you move that waiting into a send helper that only waits before sending?

```ts
private async socketSend(data: string): Promise<void> {
  if (!this.socket) {
    throw new Error("No socket available");
  }
  await this.socket.connected;
  await this.socket.send(data);
}
```

This ensures the socket is connected also when we use the constructor directly.